File: src/tools/principles.ts
import type { RunOpts } from "../registry.js";
import type { HealthStatus, RepairResult } from "../util/health.js";

export async function install(_opts: RunOpts): Promise<boolean> {
  // Instruction-only tool — no install needed
  return true;
}

export function installedVersion(): string | null {
  return null;
}

export async function latestVersion(): Promise<string | null> {
  return null;
}

export function healthCheck(): HealthStatus {
  return { healthy: true, version: null, issues: [] };
}

export async function repair(_opts: RunOpts): Promise<RepairResult> {
  const h = healthCheck();
  return { success: true, message: "Principles is instruction-only", healthAfterRepair: h };
}

File: src/tools/principles.rs
use crate::registry::RunOpts;
use crate::tools::Tool;
use crate::util::errors::Result;
use crate::util::health::{HealthStatus, RepairResult};

pub struct PrinciplesTool;

impl Tool for PrinciplesTool {
    async fn install(&self, _opts: &RunOpts) -> Result<bool> {
        Ok(true)
    }

    fn installed_version(&self) -> Option<String> {
        installed_version()
    }

    async fn latest_version(&self) -> Result<Option<String>> {
        Ok(None)
    }

    fn health_check(&self) -> HealthStatus {
        health_check()
    }
}

pub fn installed_version() -> Option<String> {
    Some("instruction-only".to_string())
}

pub fn health_check() -> HealthStatus {
    HealthStatus {
        healthy: true,
        version: None,
        issues: vec![],
    }
}

pub async fn repair(_opts: &RunOpts) -> RepairResult {
    let h = health_check();
    RepairResult {
        success: true,
        message: "Principles is instruction-only".to_string(),
        health_after_repair: Some(h),
    }
}
